upstream: yes

From c1324f186bc526aed7b3a63dfe514d34870548fb Mon Sep 17 00:00:00 2001
From: Kevin Pulo <kev@pulo.com.au>
Date: Tue, 19 Nov 2019 21:49:48 +1100
Subject: [PATCH] Avoid unnecessary ncurses clear() of the screen.

Instead, just erase() the screen, so that it can be drawn over again and
then flushed at refresh() time.

This fixes the problem where redraws in a slow terminal (eg. mosh or
over a slow ssh link) sometimes render the cleared screen, causing a
distracting visible "flash".
---
 src/draw.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/draw.cpp b/src/draw.cpp
index 1ae1eb8..ec66a56 100644
--- a/src/draw.cpp
+++ b/src/draw.cpp
@@ -829,7 +829,7 @@ void NCursesInterface::doRender()
 
 void NCursesInterface::doRedraw()
 {
-    clear();
+    erase();
     doRender();
     refresh();
 }
-- 
2.30.0

